internals/controllers/cappPlacementSync_controller.go

```diff
@@ -76,37 +77,40 @@ var CappPredicateFuncs = predicate.Funcs{
 
 // SyncManifestWork checks whether the manifest work deploying the service exists in the managed cluster namespace
 // If it does, it updates the service in the manifest work spec, if it doesn't, it creates it
-func (r *ServiceNamespaceReconciler) SyncManifestWork(capp rcsv1alpha1.Capp, ctx context.Context, l logr.Logger) (ctrl.Result, error) {
+func (r *ServiceNamespaceReconciler) SyncManifestWork(capp rcsv1alpha1.Capp, ctx context.Context, logger logr.Logger) (ctrl.Result, error) {
 	mwName := utils.NamespaceManifestWorkPrefix + capp.Namespace + "-" + capp.Name
 	managedClusterName := capp.Annotations[utils.AnnotationKeyHasPlacement]
 	var mw workv1.ManifestWork
-	manifests, err := utils.GatherCappResources(capp, ctx, l, r.Client)
+	manifests, err := utils.GatherCappResources(capp, ctx, logger, r.Client)
 	if err != nil {
-		statusutils.SetVolumesCondition(capp, ctx, r.Client, l, false, err.Error())
-		return ctrl.Result{}, err
+		r.EventRecorder.Event(&capp, eventTypeError, eventCappVolumeNotFound, err.Error())
+		statusutils.SetVolumesCondition(capp, ctx, r.Client, logger, false, err.Error())
+		return ctrl.Result{}, fmt.Errorf("Failed to get one of the volumes from capp spec: %s", err.Error())
 	}
 
 	if err := r.Get(ctx, types.NamespacedName{Name: mwName, Namespace: managedClusterName}, &mw); err != nil {
 		if errors.IsNotFound(err) {
 			mw := utils.GenerateManifestWorkGeneric(mwName, managedClusterName, manifests, workv1.ManifestConfigOption{})
 			utils.SetManifestWorkCappAnnotations(*mw, capp)
 			if err := r.Create(ctx, mw); err != nil {
-				return ctrl.Result{}, err
+				r.EventRecorder.Event(&capp, eventTypeError, eventCappManifestWorkCreationFailed, err.Error())
+				return ctrl.Result{}, fmt.Errorf("Failed to create ManifestWork: %s", err.Error())
 			}
+			logger.Info(fmt.Sprintf("Created ManifestWork %s for capp %s", mwName, capp.Name))
+			r.EventRecorder.Event(&capp, eventTypeNormal, eventCappManifestWorkCreated, fmt.Sprintf("created ManifestWork %s for capp %s", mwName, capp.Name))
 			return ctrl.Result{}, nil
 		}
 		return ctrl.Result{}, err
 	}
 	mw.Spec.Workload.Manifests = manifests
 
 	if err = r.Update(ctx, &mw); err != nil {
-		l.Error(err, "unable to update ManifestWork")
 		if errors.IsConflict(err) {
+			logger.Info(fmt.Sprint("conflict while updating manifestWork trying again in a few seconds"))
 			return ctrl.Result{RequeueAfter: time.Second * 2}, nil
 		}
-		return ctrl.Result{}, err
+		return ctrl.Result{}, fmt.Errorf("failed to sync manifestWork: %s", err.Error())
```

Comment:
Be consistent with the naming, there are other places where it's written as `ManifestWork` and not `manifestWork`